### ISSUE TITLE
Feature Request: Rule to detect parameterless `takeUntilDestroyed()` usage outside injection context

### ISSUE THREAD

**Issue by askeric2** [angular-eslint/angular-eslint#2706]
**Problem:**
`takeUntilDestroyed()` can be called without parameters only within an injection context (constructor, field initializers). When called without a `DestroyRef` parameter outside injection context (e.g., in `ngOnInit()`), it throws a runtime error: `NG0203: takeUntilDestroyed() can only be used within an injection context`.

**Proposed Solution:**
A lint rule that warns when `takeUntilDestroyed()` is called without parameters outside of an injection context.

**Examples:**

❌ **Should warn:**
```typescript
export class MyComponent implements OnInit {
  ngOnInit() {
    this.obs$.pipe(takeUntilDestroyed()).subscribe(); // Runtime error!
  }
}
```

✅ **Should pass:**
```typescript
export class MyComponent implements OnInit {
  private destroyRef = inject(DestroyRef);
  
  ngOnInit() {
    this.obs$.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(); // OK
  }
}
```

✅ **Should pass:**
```typescript
export class MyComponent {
  // Field initializer is injection context
  data$ = this.source$.pipe(takeUntilDestroyed()); // OK
}
```

**Related Work:**
This is related to issue #1400 which proposes detecting `inject()` usage outside injection contexts. The same injection context detection logic could potentially be shared.

**Implementation Considerations:**
As discussed in #1400, detecting injection context is complex because it's not limited to constructors and field initializers. Injection context also includes:
- HTTP interceptors
- Route guards
- Resolvers  
- App initializers
- Environment/Platform initializers
- Other framework-provided injection contexts

**Suggested Approach:**
Given this complexity, this rule could be implemented in phases:

**Phase 1 (Simpler):** Detect obvious violations in lifecycle hooks
- `ngOnInit`, `ngAfterViewInit`, `ngOnDestroy`, etc.
- These are clearly NOT injection contexts and are common mistake locations

**Phase 2 (Advanced):** Expand to detect usage in regular methods
- Would require more sophisticated analysis

Even a Phase 1 implementation would catch the majority of common mistakes and provide significant value.

**Benefits:**
- Catches common mistakes at lint time instead of runtime
- Improves developer experience
- Prevents `NG0203` errors

**Comment by reduckted** [comment]
https://github.com/angular-eslint/angular-eslint/discussions/2672

**Comment by JamesHenry** [comment]
Closing as a duplicate of https://github.com/angular-eslint/angular-eslint/discussions/2672 - as already stated on there, PRs Welcome! 🙏
